docs: What’s New & Upgrade Guide 10.1 (#70636)
grafana/grafana: docs/sources/whatsnew/whats-new-in-v10-1.md

{{< figure src="/media/docs/grafana/data-sources/loki-shortlink.png" max-width="750px" caption="New Loki step editor" >}}

### TraceQL response streaming in Tempo

<!-- André Pereira -->

_Generally available in all editions of Grafana._

Grafana's Tempo data source latest upgrade includes support for streaming responses of TraceQL queries. With this feature, you can now see partial query results as they come in, so no more waiting for the whole query to finish. This is perfect for those long queries that take a long time to return a response.

To use this feature, toggle on the "Stream response" option in either the Search or TraceQL query type, and you'll get immediate visibility of incoming traces on the results table. This smooth integration makes data exploration a breeze and speeds up decision-making.

{{< video-embed src="/media/docs/grafana/data-sources/tempo-streaming.mp4" >}}

> <!-- Matías Wenceslao Chomicki -->
> 
> ### Configuration page redesign for Loki and Elasticsearch
> 
> Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to configuration pages and tooltips to assist you with configuring and customizing data sources.
> 
> ### Loki query splitting
> 
> In response to different query performance scenarios, we implemented query splitting, where queries that request more than a day of data are split in sub-requests of 1 day duration each. For example, requesting 7 days of logs will produce 7 requests of 1 day.
> 
> ## Explore
> 
> <!-- Matías Wenceslao Chomicki -->
> 
> ### Logs: Log rows menu when using displayed fields
> 
> When you're browsing logs in Explore you can use the Log Details component, that is displayed when you click on a row, to replace the log lines contents with the value of one or more of the log fields or labels by using the "eye" icon. When this feature is in use, you now have access to the menu displayed on mouse-over with options such as show context (if available), copy log to clipboard, or copy shortlink.
> 
> ### Logs: Improved rendering performance of log lines
> 

Reviewer: ```suggestion
### Configuration page redesign for Loki and Elasticsearch

<!-- Matías Wenceslao Chomicki -->

Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to configuration pages and tooltips to assist you with configuring and customizing data sources.

### Loki query splitting

<!-- Matías Wenceslao Chomicki -->

In response to different query performance scenarios, we implemented query splitting, where queries that request more than a day of data are split in sub-requests of 1 day duration each. For example, requesting 7 days of logs will produce 7 requests of 1 day.

## Explore

### Logs: Log rows menu when using displayed fields

<!-- Matías Wenceslao Chomicki -->

When you're browsing logs in Explore you can use the Log Details component, that is displayed when you click on a row, to replace the log lines contents with the value of one or more of the log fields or labels by using the "eye" icon. When this feature is in use, you now have access to the menu displayed on mouse-over with options such as show context (if available), copy log to clipboard, or copy shortlink.

### Logs: Improved rendering performance of log lines

<!-- Matías Wenceslao Chomicki -->

```

_Generally available in all editions of Grafana._

Grafana's Tempo data source latest upgrade includes support for streaming responses of TraceQL queries. With this feature, you can now see partial query results as they come in, so no more waiting for the whole query to finish. This is perfect for those long queries that take a long time to return a response.

To use this feature, toggle on the "Stream response" option in either the Search or TraceQL query type, and you'll get immediate visibility of incoming traces on the results table. This smooth integration makes data exploration a breeze and speeds up decision-making.

{{< video-embed src="/media/docs/grafana/data-sources/tempo-streaming.mp4" >}}

### Configuration page redesign for Loki and Elasticsearch

<!-- Matías Wenceslao Chomicki -->

Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to configuration pages and tooltips to assist you with configuring and customizing data sources.

### Loki query splitting

<!-- Matías Wenceslao Chomicki -->

> In response to different query performance scenarios, we implemented query splitting, where queries that request more than a day of data are split in sub-requests of 1 day duration each. For example, requesting 7 days of logs will produce 7 requests of 1 day.

Reviewer: ```suggestion
In response to different query performance scenarios, we implemented query splitting, where queries that request more than a day of data are split in sub-requests of one day duration each. For example, requesting seven days of logs will produce seven requests of one day.
```

This change impacts `GitHub` OAuth, `Gitlab` OAuth, `Okta` OAuth and `Generic` OAuth.

Currently if no organization role mapping is found for a user when connecting using OAuth, Grafana doesn’t update the user’s organization role.

With Grafana 10.1, on every login, if the `role_attribute_path` property does not return a role, then the user is assigned the role specified by the `auto_assign_org_role` option or the default role for the organization, by default, Viewer.

To avoid overriding manually set roles, enable the `skip_org_role_sync` option in the Grafana configuration for your OAuth provider before the user logs in for the first time.

### Preventing manual role updates for externally synced roles

<!-- Ieva Vasiljeva, AuthNZ -->

_Generally available in Grafana Open Source, Enterprise, and Cloud._

This change impacts all instances that use an external authentication provider and have [role mapping]({{< relref "../setup-grafana/configure-security/planning-iam-strategy/#role-sync" >}}) enabled.

Currently, it is possible to manually update a user's organization role (Viewer, Editor or Admin) even if this role is managed by an external authentication provider.

> This means that roles can be manually set for the duration of a user's session, but are overridden by the external authentication provider the next time when the user logs in.
> If `onlyExternalOrgRoleSync` feature toggle is enabled, manual role updates for externally managed roles are not allowed.

Reviewer: ```suggestion
This means that roles can be manually set for the duration of a user's session, but are overridden by the external authentication provider the next time the user logs in.
If the `onlyExternalOrgRoleSync` feature toggle is enabled, manual role updates for externally managed roles are not allowed.
```

{{% /admonition %}}
-->

## Dashboards and visualizations

### Disconnect values in time series, trend, and state timeline visualizations

_Generally available in all editions of Grafana._

<!-- Nathan Marrs -->

You can now choose whether to set a threshold above which values in the data should be disconnected. This can be useful in cases where you have sensors that report a value at a set interval, but you want to disconnect the values when the sensor does not respond. This feature complements the existing [connect null values functionality]({{< relref "../panels-visualizations/visualizations/time-series/#connect-null-values" >}}).

To learn more, refer to our [disconnect values documentation]({{< relref "../panels-visualizations/visualizations/time-series/#disconnect-values" >}}).

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-disconnect-values-examples.png" max-width="750px" caption="Disconnect values in time series, trend, and state timeline visualizations" >}}

### Flamegraph improvements

_Generally available in all editions of Grafana._

<!-- Andrej Ocenas -->

We have added 4 new features to the flamegraph visualization:

> - **Sandwich view**: You can now show a sandwich of any symbol in the flamegraph. Sandwich view will show all the callers on the top and all the callees of the symbol on the bottom. This is useful when you want to see the context of a symbol.

Reviewer: ```suggestion
- **Sandwich view**: You can now show a sandwich of any symbol in the flame graph. Sandwich view shows all the callers on the top and all the callees of the symbol on the bottom. This is useful when you want to see the context of a symbol.
```

> - **Switching color scheme**: You can now switch color scheme between color gradient by the relative value of a symbol or by package name of a symbol.
> - **Switching symbol name alignment**: With symbols with long name it may be problematic to differentiate them if they have the same prefix. This new option allows you to align the text to left or right so that you can see the part of the symbol name that is important.
> - **Improved navigation**: You can also highlight a symbol or switch on sandwich view for a symbol from the table. Also, a new status bar on top of the flamegraph gives you an overview of which views are enabled.
> 
> {{< video-embed src="/media/docs/grafana/panels-visualizations/screen-recording-grafana-10.1-flamegraph-whatsnew.mp4" >}}

Reviewer 2: ```suggestion
- **Switching color scheme**: You can now switch color scheme between color gradient by the relative value of a symbol or by package name of a symbol.
- **Switching symbol name alignment**: With symbols with long name it may be problematic to differentiate them if they have the same prefix. This new option allows you to align the text to left or right so that you can see the part of the symbol name that is important.
- **Improved navigation**: You can also highlight a symbol or switch on sandwich view for a symbol from the table. Also, a new status bar on top of the flame graph gives you an overview of which views are enabled.

{{< video-embed src="/media/docs/grafana/panels-visualizations/screen-recording-grafana-10.1-flamegraph-whatsnew.mp4" >}}
```

### Transformations redesign

> The transformations tab has gotten an improved user experience and visual redesign! Now you can view transformations with categories and illustrations.

Reviewer: ```suggestion
The transformations tab has gotten an improved user experience and visual redesign. Now you can view transformations with categories and illustrations.
```

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-transformations.png" max-width="750px" caption="Transformations redesign" >}}

### Format Time Transformation (Alpha)

<!-- Kyle Cunningham -->

_Generally available in all editions of Grafana._

> When working with times it can be useful to have different time formats. The new format time transformation allows any time format supported by Moment.js to be used in displaying times. When used in conjunction with the _Group by Value_ transformation this can also be used to bucket days, weeks, and other time windows together.

Reviewer 2: ```suggestion
When working with times it can be useful to have different time formats. The new format time transformation allows any time format supported by Moment.js to be used in displaying times. When used in conjunction with the _Group by Value_ transformation, this can also be used to bucket days, weeks, and other time windows together.
```

### Logs: Log rows menu when using displayed fields

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana._

When you're browsing logs you can use the Log Details component, that is displayed when you click on a row, to replace the log lines contents with the value of one or more of the log fields or labels by using the "eye" icon. When this feature is in use, you now have access to the menu displayed on mouse-over with options such as show context (if available), copy log to clipboard, or copy shortlink.

### Logs: Improved rendering performance of log lines

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana._

With Grafana 10.1 browsing log lines is faster than ever before after a series of performance optimizations done for log-related components.

### Logs: See more log lines in logs context

<!-- Gabor Farkas, Sven Grossmann -->

_Generally available in all editions of Grafana._

> Log context allows you to view additional lines surrounding a specific log entry. With this enhancement, you can access as many log lines as needed within the log context. As you scroll through the logs, Grafana dynamically loads more log lines, ensuring a seamless and continuous viewing experience.

Reviewer 2: ```suggestion
Log context allows you to view additional lines surrounding a specific log entry. With this enhancement, you can access as many log lines as needed within the log context. Grafana dynamically loads more log lines as you scroll through the logs, ensuring a seamless and continuous viewing experience.
```

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-heatmap-datalinks.png" max-width="750px" caption="Heatmap datalink support" >}}

## Data sources

### Step editor in Loki

<!-- Ivana Huckova -->

_Generally available in all editions of Grafana._

We've improved the Loki query editor by adding a new step editor. This editor allows you to specify a value for the _step_ parameter in Loki queries. You can use this parameter when making metric queries to Loki or when you want to get a matrix response from your queries.

By default, the step parameter is set to the value of the `$__interval` variable. This variable is calculated based on the time range and the width of the graph (in pixels). If you want to learn more about the Loki step parameter, you can visit [the Loki step parameter documentation](<(https://grafana.com/docs/loki/latest/api/#step-versus-interval)>).

{{< figure src="/media/docs/grafana/data-sources/loki-step-editor.png" max-width="750px" caption="New Loki step editor" >}}

### Copy link to a Loki log line

<!-- Sven Grossmann -->

_Generally available in all editions of Grafana._

> A new linking of Loki log lines in Explore allows you to quickly navigate to specific log entries for precise analysis. By clicking the **Copy shortlink** button for a log line, you can generate and copy a [short URL]({{< relref "../developers/http_api/short_url/" >}}) that provides direct access to the exact log entry within an absolute time range. When you open the link, Grafana automatically scrolls to the corresponding log line and highlights it with a blue background, making it easy to identify and focus on the relevant information.

Reviewer: ```suggestion
A new linking of Loki log lines in Explore allows you to quickly navigate to specific log entries for precise analysis. By clicking the **Copy shortlink** button for a log line, you can generate and copy a [short URL]({{< relref "../developers/http_api/short_url/" >}}) that provides direct access to the exact log entry within an absolute time range. When you open the link, Grafana automatically scrolls to the corresponding log line and highlights it with a blue background, making it easy to identify and focus on the relevant information.
```

{{< figure src="/media/docs/grafana/data-sources/loki-shortlink.png" max-width="750px" caption="New Loki step editor" >}}

### TraceQL response streaming in Tempo

<!-- André Pereira -->

_Generally available in all editions of Grafana._

> Grafana's Tempo data source latest upgrade includes support for streaming responses of TraceQL queries. With this feature, you can now see partial query results as they come in, so no more waiting for the whole query to finish. This is perfect for those long queries that take a long time to return a response.

Reviewer: ```suggestion
Grafana's Tempo data source latest upgrade includes support for streaming responses of PromQL queries. With this feature, you can now see partial query results as they come in, so no more waiting for the whole query to finish. This is perfect for those long queries that take a long time to return a response.
```

To use this feature, toggle on the "Stream response" option in either the Search or TraceQL query type, and you'll get immediate visibility of incoming traces on the results table. This smooth integration makes data exploration a breeze and speeds up decision-making.

{{< video-embed src="/media/docs/grafana/data-sources/tempo-streaming.mp4" >}}

### Span filtering for traces is GA

_Generally available in all editions of Grafana._

<!-- Joey Tawadrous -->

Since the last release, we've made several improvements to span filtering. We're promoting span filters out of public preview and into general availability.

> Span filters allow you to work much more efficiently with traces that consist of a large number of spans.
> 
> Span filters exist above the trace view and allow you to filter the spans that are shown in the trace view. The more filters you add, the more specific are the filtered spans.

Reviewer: ```suggestion
Span filters allow you to work much more efficiently with traces that consist of a large number of spans.
Span filters exist above the trace view and allow you to filter the spans that are shown in the trace view. The more filters you add, the more specific are the filtered spans.
```

---
description: Feature and improvement highlights for Grafana v10.1
keywords:
  - grafana
  - new
  - documentation
  - '10.1'
  - release notes
labels:
products:
  - cloud
  - enterprise
  - oss
title: What's new in Grafana v10.1
weight: -38
---

# What’s new in Grafana v10.1

> Welcome to Grafana 10.1! Read on to learn about changes to dashboards and visualizations, data sources, and security and authentication and more. We're particularly excited about a set of improvements to visualizing logs from [Loki](https://grafana.com/products/cloud/logs/) and other logging data sources in Explore mode, and our Flame graph panel, used to visualize profiling data from [Pyroscope](https://grafana.com/blog/2023/03/15/pyroscope-grafana-phlare-join-for-oss-continuous-profiling/?pg=oss-phlare&plcmt=top-promo-banner) and other continuous profiling data sources.

Reviewer: ```suggestion
Welcome to Grafana 10.1! Read on to learn about changes to dashboards and visualizations, data sources, security and authentication and more. We're particularly excited about a set of improvements to visualizing logs from [Loki](https://grafana.com/products/cloud/logs/) and other logging data sources in Explore mode, and our Flame graph panel, used to visualize profiling data from [Pyroscope](https://grafana.com/blog/2023/03/15/pyroscope-grafana-phlare-join-for-oss-continuous-profiling/?pg=oss-phlare&plcmt=top-promo-banner) and other continuous profiling data sources.
```

{{< video-embed src="/media/docs/grafana/panels-visualizations/screen-recording-grafana-10.1-flamegraph-whatsnew.mp4" >}}

### Distinguish widgets from visualizations

<!-- Alexa Vargas, Juan Cabanas -->

_Experimental in all editions of Grafana_

This experimental feature introduces a clear distinction between two different categories of panel plugin types: visualization panels that consume a data source and a new type, called _widgets_, that don't require a data source.

Now, you can easily add widgets like Text, News, and Annotation list without the need to select a data source. The plugins list and library panels are filtered based on whether you've selected a widget or visualization, providing a streamlined editing experience.

To see the widget editor in Grafana OSS or Enterprise, enable the `vizAndWidgetSplit` feature toggle. If you’re using Grafana Cloud and would like to enable this feature, please contact customer support.

{{< figure src="/media/docs/grafana/dashboards/WidgetVizSplit.png" max-width="750px" caption="New widget option added to empty dashboards" >}}

### Transformations redesign

> 

Reviewer: ```suggestion

<!-- Catherine Gui -->
```

{{< video-embed src="/media/docs/grafana/panels-visualizations/screen-recording-grafana-10.1-flamegraph-whatsnew.mp4" >}}

### Distinguish widgets from visualizations

<!-- Alexa Vargas, Juan Cabanas -->

_Experimental in all editions of Grafana_

This experimental feature introduces a clear distinction between two different categories of panel plugin types: visualization panels that consume a data source and a new type, called _widgets_, that don't require a data source.

Now, you can easily add widgets like Text, News, and Annotation list without the need to select a data source. The plugins list and library panels are filtered based on whether you've selected a widget or visualization, providing a streamlined editing experience.

To see the widget editor in Grafana OSS or Enterprise, enable the `vizAndWidgetSplit` feature toggle. If you’re using Grafana Cloud and would like to enable this feature, please contact customer support.

{{< figure src="/media/docs/grafana/dashboards/WidgetVizSplit.png" max-width="750px" caption="New widget option added to empty dashboards" >}}

### Transformations redesign

<!-- Catherine Gui -->

> 

Reviewer: @catherineymgui - What's the availability of this feature? Experimental, public preview, private preview, or GA? And is it available in OSS, Enterprise, Cloud, or all three?

> The transformations tab has an improved user experience and visual redesign. Now, you can explore transformations with categories and illustrations.

Reviewer: @catherineymgui - I found the second sentence a bit unclear. Can we go with this? 

**Note, please don't commit any changes without checking in with Quynton first.**

```suggestion
The transformations tab has an improved user experience and visual redesign. Now, transformations are categorized, and each transformation type has an illustration to help you choose the right one.
```

Reviewer 3: This sounds good to me.

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-transformations.png" max-width="750px" caption="Transformations redesign" >}}

### Format Time transformation

<!-- Kyle Cunningham -->

_Available in public preview in all editions of Grafana_

> When working with date and time data, it can be useful to have different time formats. With the new Format Time transformation, you can convert any time format to any other one supported by Moment.js. When used in conjunction with the Group by Value transformation, this can also be used to bucket days, weeks, and other time windows together.

Reviewer: @codeincarnate - Do you mean the Group by transformation with the selection Value? I want to make sure we're not priming a user to look for something in the UI (a Group by Value option) that they aren't going to see? If so, I'll tweak this upon my return. 

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 4: @imatwawana you're correct on the naming. I'll check in with Quynton about this. Thanks! 🙏

Reviewer 4: Just to make sure relevant changes are here https://github.com/grafana/grafana/commit/1c81910f5328c2d372ee61aea608cb253e8a6644

Reviewer 4: Ran the change noted past Quynton and got his go ahead.  I've added the reference commit to this branch 😄

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-heatmap-datalinks.png" max-width="750px" caption="Heatmap datalink support" >}}

## Data sources

### Step editor in Loki

<!-- Ivana Huckova -->

_Generally available in all editions of Grafana_

We've improved the Loki query editor by adding a new **Step** editor field. This editor allows you to specify a value for the `step` parameter in Loki queries. You can use this parameter when making metric queries to Loki or when you want a matrix response from your queries.

By default, the `step` parameter is set to the value of the `$__interval` variable. This variable is calculated based on the time range and the width of the graph (in pixels). If you want to learn more about the Loki `step` parameter, you can visit [the Loki step parameter documentation](/docs/loki/latest/api/#step-versus-interval).

{{< figure src="/media/docs/grafana/data-sources/loki-step-editor.png" max-width="750px" caption="New Loki step editor" >}}

### Copy link to a Loki log line

<!-- Sven Grossmann -->

_Generally available in all editions of Grafana_

> A new linking of Loki log lines in Explore allows you to quickly navigate to specific log entries for precise analysis. By clicking the **Copy shortlink** button for a log line, you can generate and copy a [short URL]({{< relref "../developers/http_api/short_url/" >}}) that provides direct access to the exact log entry within an absolute time range. When you open the link, Grafana automatically scrolls to the corresponding log line and highlights it with a blue background, making it easy to identify and focus on the relevant information.

Reviewer: @svennergr - Is the background blue in light mode because it doesn't appear blue in dark mode? Can we just take out the colour and just say it'll be highlighted? 

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 5: It should be blue in both modes. E.g. https://ops.grafana-ops.net/goto/v87rFd64R?orgId=1

<img width="519" alt="image" src="https://github.com/grafana/grafana/assets/8092184/f5b64712-72f6-4136-9825-616d77e4a0dc">

Reviewer 5: Nevertheless I am also fine with removing the color from the text.

_Generally available in all editions of Grafana_

<!-- Joey Tawadrous -->

Since the last release, we've made several improvements to span filtering. Now, we're promoting span filters out of public preview and into general availability.

Span filters allow you to work much more efficiently with traces that consist of a large number of spans. Span filters exist above the trace view and allow you to filter the spans that are shown in the trace view. The more filters you add, the more specific the filtered spans.

Currently, you can add one or more of the following filters:

- Service Name
- Span Name
- Duration
- Tags (which includes tags, process tags, and log fields)

To only show the spans you've matched, you can enable the **Show matches only** toggle.

Learn more about span filtering in our [Tempo data source documentation]({{< relref "../datasources/tempo/#span-filters" >}}).

{{< figure src="/media/docs/tempo/screenshot-grafana-tempo-span-filters-v10-1.png" max-width="750px" caption="Traces span filtering" >}}

### Configuration page redesign for Loki and Elasticsearch

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

> The Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to configuration pages, as well as tooltips to assist you with configuring and customizing data sources.

Reviewer: @matiasb - In the second sentence, it sounds like the configuration pages in Grafana for these data sources have new subsections with links to configuration pages. Where are these other configuration pages? Or are the subsections and links not on the data source configuration pages. 

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 6: Oh my bad. It's actually "with links to documentation pages".

### Loki query splitting

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

In response to different query performance scenarios, we've implemented query splitting. Now, queries that request more than a day of data are split into sub-requests, each with a duration of one day. For example, requesting seven days of logs will produce seven requests of one day.

### Easier to use query editor for Elasticsearch

<!-- Gabor Farkas -->

_Generally available in all editions of Grafana_

The Elasticsearch query editor now allows convenient switching between logs, metrics, and raw data directly from the top, eliminating the need to go through the metric selector.

### Metrics explorer

> 

Reviewer: @catherineymgui - What's the availability of this feature—Experimental, private preview, public preview or GA? And is it in OSS, Enterprise, Cloud or all three?

Reviewer 3: GA - and all 3.

> 

Reviewer: ```suggestion

<!-- Catherine Gui -->

```

The Metrics explorer is a new feature that enhances metrics browsing in the Prometheus query builder. The Metrics explorer makes it easier for you to find the right metric, and get comfortable with PromQL. You can now explore metrics with additional metadata, perform fuzzy search on the metric name or description, and filter on the Prometheus type.

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-metrics-explorer.png" max-width="750px" caption="Searching in Metrics explorer" >}}

> 

Reviewer: @iwysiu - Should the next three entries be under the Data sources section of the What's new?

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 7: Yes, they should be. Are they not?

Reviewer: They are but the Redshift & Athena items are at the same level as the Data Sources heading, so it looks like they're their own sections. I'll fix the heading levels.

## Redshift and Athena: Async query data support

<!-- Isabella Siu, Kevin Yu, Andrés Martínez -->

_Generally available in all editions of Grafana_

Async query data support in Redshift and Athena makes queries over multiple requests (starting, checking status, and fetching the results) instead of in a single request query. This is useful for queries that can potentially run for a long time and time out. This feature was previously available behind a feature toggle and is now be generally available and enabled by default.

## Redshift and Athena: Async query caching

<!-- Isabella Siu -->

_Experimental in Grafana Enterprise, Cloud Pro, and Cloud Advanced_

This feature adds support for query caching of async queries in the Athena and Redshift data source plugins. To try this feature, enable both the `useCachingService` and `awsAsyncQueryCaching` feature toggles. If you’re using Grafana Cloud and would like to enable this experimental feature, please contact customer support.

### CloudWatch logs Monaco query editor

> 

Reviewer: @iwysiu - Is this section meant to be a child of the **Redshift and Athena: Async query caching item**? Or just an item under Data sources?

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 7: It should be an item under data sources

<!-- Isabella Siu, Kevin Yu -->

_Experimental in all editions of Grafana_

The CloudWatch logs query editor is moving from being a Slate-based editor to a Monaco-based editor. This new Monaco-based editor will provide improved syntax highlighting, and auto-completion. To use the Monaco-based query editor, enable the `cloudWatchLogsMonacoEditor` feature toggle. If you’re using Grafana Cloud and would like to enable this feature, please contact customer support.

### InfluxDB Backend Mode

> 

Reviewer: @itsmylife - Should this section be under **Data sources** rather than a child of the the **Redshift and Athena: Async query caching** section?

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer: @itsmylife - What is the availability of this feature? Experimental, private preview, public preview, or GA? And is it in available in Grafana OSS, Enterprise, Cloud or all three?

Also, I've made some edits to this entry. Please let me know if they're okay. 

**Note, please don't commit any changes without checking in with Quynton first.**

Reviewer 8: 1. Yes it should be under data sources. I thought I put it there. 
2. Both
3. They are perfect. Thanks.

Reviewer: There was a nesting issue with the Redshift and Athena entries, so I just wanted to make sure of where this should live. Thanks!

#### UI changes

- Added an **Angular** badge next to affected plugins in the plugins catalog.
- Added an alert at the top of a plugin's page in the plugins catalog when browsing Angular plugins.
- Added an alert at the top of the query editor when editing panels that use Angular data source plugins.

#### Other changes

- Angular Plugins will not be loaded if [angular_support_enabled]({{< relref "../setup-grafana/configure-grafana/#angular_support_enabled" >}}) is set to `false`.

Learn more in our [Angular support deprecation documentation]({{< relref "../developers/angular_deprecation/" >}}).

### Deprecated provisioning of data sources with invalid UIDs

<!-- Giuseppe Guerra, Plugins Platform -->

_Generally available in all editions of Grafana_

Grafana now logs an error when provisioning data sources with invalid UIDs. A valid UID is a combination of a-z, A-Z, 0-9 (alphanumeric), `-` (dash), and `_` (underscore), with a maximum length of 40 characters.

Provisioning data sources with invalid UIDs will be removed in future versions of Grafana, and will return an error instead.

## Subfolders: folder selection

> 

Reviewer: @zsofiakm - Should this be under **Dashboards and visualizations**?

**Note, please don't commit any changes without checking in with Quynton first.**

<!-- Zsofia Komaromi -->

_Available in public preview in all editions of Grafana_

When saving or moving a dashboard, you can now see the full folder tree when selecting the destination folder.

To get started creating subfolders, enable the `nestedFolders` feature toggle. We recommend that you enable this feature only on test or development instances, rather than in production environments.

{{< figure src="/media/docs/grafana/screenshot-grafana-10.1-subfolders-folder-picker.png" max-width="750px" caption="Selecting a folder in Grafana" >}}

## Activate draft reports

> 

Reviewer: @rhorvath - Should this be under Dashboards and visualizations?

**Note, please don't commit any changes without checking in with Quynton Johnson first.**

Reviewer 9: Good point, yes, we can move this under Dashboards section

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-heatmap-datalinks.png" max-width="750px" caption="Heatmap datalink support" >}}

## Data sources

### Step editor in Loki

<!-- Ivana Huckova -->

_Generally available in all editions of Grafana_

We've improved the Loki query editor by adding a new **Step** editor field. This editor allows you to specify a value for the `step` parameter in Loki queries. You can use this parameter when making metric queries to Loki or when you want a matrix response from your queries.

By default, the `step` parameter is set to the value of the `$__interval` variable. This variable is calculated based on the time range and the width of the graph (in pixels). If you want to learn more about the Loki `step` parameter, you can visit [the Loki step parameter documentation](/docs/loki/latest/api/#step-versus-interval).

{{< figure src="/media/docs/grafana/data-sources/loki-step-editor.png" max-width="750px" caption="New Loki step editor" >}}

### Copy link to a Loki log line

<!-- Sven Grossmann -->

_Generally available in all editions of Grafana_

> A new linking of Loki log lines in Explore allows you to quickly navigate to specific log entries for precise analysis. By clicking the **Copy shortlink** button for a log line, you can generate and copy a [short URL]({{< relref "../developers/http_api/short_url/" >}}) that provides direct access to the exact log entry within an absolute time range. When you open the link, Grafana automatically scrolls to the corresponding log line and highlights it with a blue background, making it easy to identify and focus on the relevant information.

Reviewer: ```suggestion
A new linking of Loki log lines in Explore allows you to quickly navigate to specific log entries for precise analysis. By clicking the **Copy shortlink** button for a log line, you can generate and copy a [short URL]({{< relref "../developers/http_api/short_url/" >}}) that provides direct access to the exact log entry within an absolute time range. When you open the link, Grafana automatically scrolls to the corresponding log line and highlights it, making it easy to identify and focus on the relevant information.
```

_Generally available in all editions of Grafana_

<!-- Joey Tawadrous -->

Since the last release, we've made several improvements to span filtering. Now, we're promoting span filters out of public preview and into general availability.

Span filters allow you to work much more efficiently with traces that consist of a large number of spans. Span filters exist above the trace view and allow you to filter the spans that are shown in the trace view. The more filters you add, the more specific the filtered spans.

Currently, you can add one or more of the following filters:

- Service Name
- Span Name
- Duration
- Tags (which includes tags, process tags, and log fields)

To only show the spans you've matched, you can enable the **Show matches only** toggle.

Learn more about span filtering in our [Tempo data source documentation]({{< relref "../datasources/tempo/#span-filters" >}}).

{{< figure src="/media/docs/tempo/screenshot-grafana-tempo-span-filters-v10-1.png" max-width="750px" caption="Traces span filtering" >}}

### Configuration page redesign for Loki and Elasticsearch

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

> The Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to configuration pages, as well as tooltips to assist you with configuring and customizing data sources.

Reviewer: ```suggestion
The Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to documentation pages, as well as tooltips to assist you with configuring and customizing data sources.
```

### Loki query splitting

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

In response to different query performance scenarios, we've implemented query splitting. Now, queries that request more than a day of data are split into sub-requests, each with a duration of one day. For example, requesting seven days of logs will produce seven requests of one day.

### Easier to use query editor for Elasticsearch

<!-- Gabor Farkas -->

_Generally available in all editions of Grafana_

The Elasticsearch query editor now allows convenient switching between logs, metrics, and raw data directly from the top, eliminating the need to go through the metric selector.

### Metrics explorer

<!-- Catherine Gui -->

> 

Reviewer: ```suggestion

_Generally available in all editions of Grafana_

```

The Metrics explorer is a new feature that enhances metrics browsing in the Prometheus query builder. The Metrics explorer makes it easier for you to find the right metric, and get comfortable with PromQL. You can now explore metrics with additional metadata, perform fuzzy search on the metric name or description, and filter on the Prometheus type.

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-metrics-explorer.png" max-width="750px" caption="Searching in Metrics explorer" >}}

> ## Redshift and Athena: Async query data support

Reviewer: ```suggestion
### Redshift and Athena: Async query data support
```

<!-- Isabella Siu, Kevin Yu, Andrés Martínez -->

_Generally available in all editions of Grafana_

Async query data support in Redshift and Athena makes queries over multiple requests (starting, checking status, and fetching the results) instead of in a single request query. This is useful for queries that can potentially run for a long time and time out. This feature was previously available behind a feature toggle and is now be generally available and enabled by default.

> ## Redshift and Athena: Async query caching

Reviewer: ```suggestion
### Redshift and Athena: Async query caching
```

<!-- Isabella Siu -->

_Experimental in Grafana Enterprise, Cloud Pro, and Cloud Advanced_

This feature adds support for query caching of async queries in the Athena and Redshift data source plugins. To try this feature, enable both the `useCachingService` and `awsAsyncQueryCaching` feature toggles. If you’re using Grafana Cloud and would like to enable this experimental feature, please contact customer support.

### CloudWatch logs Monaco query editor

<!-- Isabella Siu, Kevin Yu -->

_Experimental in all editions of Grafana_

The CloudWatch logs query editor is moving from being a Slate-based editor to a Monaco-based editor. This new Monaco-based editor will provide improved syntax highlighting, and auto-completion. To use the Monaco-based query editor, enable the `cloudWatchLogsMonacoEditor` feature toggle. If you’re using Grafana Cloud and would like to enable this feature, please contact customer support.

### InfluxDB Backend Mode

<!-- Ismail Simsek -->

> 

Reviewer: ```suggestion

_Available in public preview in all editions of Grafana_

```

### Elasticsearch logs sample

<!-- Gareth Dawson -->

_Generally available in all editions of Grafana_

For Elasticsearch metric queries in Explore, you can now see the sample of log lines that contributed to the displayed results. To see these logs, click the collapsed Logs sample panel under your graph or table panel. If you want to interact with the log lines or modify the log query, click the **Open logs in split view** button and the log query will be executed in the split view.

### Panel plugins in Explore

<!-- Ben Donnelly -->

_Experimental in all editions of Grafana_

Data source plugin developers can now use any plugin to visualize data in Explore. Similar to `preferredVisualizationType`, we've introduced an experimental API to render visualizations by plugin ID. In the returned data frame, set the meta option `preferredVisualisationPluginId` to the plugin ID you want to be used when showing the data for given data frame.

## Alerting

> 

Reviewer: @brendamuir - I've made a number of suggestions to the section to make the writing style more consistent with the rest of the What's new and I think improve clarity in some cases. Please let me know if these are okay. 

**Note, please don't commit any changes without checking in with Quynton first.**

_All Alerting features are generally available in all editions of Grafana_

We’ve made a number of improvements to simplify the alert rule creation process as well as improvements to contact points and alert management. For all the details, refer to our [Alerting documentation]({{< relref "../alerting" >}}).

### Alert rules

We’ve made the following changes to alert rules.

#### Alert instance routing preview

_This feature is for Grafana-managed alert rules only._

> Preview how your alert instances will be routed if they fire while you're creating your alert rule. View routing for each Alertmanager you have configured to receive Grafana-managed alerts and if required, you can easily make adjustments to your custom labels to change the way your alert instances are routed.

Reviewer: ```suggestion
Preview how your alert instances will be routed if they fire while you're creating your alert rule. You can view routing for each Alertmanager you have configured to receive Grafana-managed alerts, and if required, easily make adjustments to your custom labels to change the way your alert instances are routed.
```

Reviewer 10: sounds good to me-feel free to commit the change.

{{< figure src="/media/docs/alerting/alert-routing-preview.png" max-width="750px" caption="Alert instance routing preview" >}}

#### Alert rule types

> Enables you to switch to a data source-managed alert rule if your data source is configured to support alert rule creation (Ruler API enabled). By default, the alert rule type is Grafana-managed.

Reviewer: ```suggestion
You can switch to a data source-managed alert rule if your data source is configured to support alert rule creation (Ruler API enabled). By default, the alert rule type is Grafana-managed.
```

Reviewer 10: sounds good to me-go ahead and make commit the change:)

{{< figure src="/media/docs/alerting/alert-rule-types.png" max-width="750px" caption="Alert rule types" >}}

#### UI improvements

- **Alert evaluation behavior**: New UI components for creating a folder and adding an evaluation group along with improved text and validation.

> - **Alert Rule list page**: the process of creating recording rules (More drop down) is now separate from Grafana-managed and data source-managed alert rules (**+New alert rule**)

Reviewer: ```suggestion
- **Alert Rule list page**: The process of creating recording rules (**More** drop down) is now separate from Grafana-managed and data source-managed alert rules (**+New alert rule**)
```

  .
- **Annotations display**: Adding a summary, description, and runbook URL as annotations are now optional. The dashboard and panel names are now also linked directly, making it easier to access.

> - **View YAML button**: Displays alert rule configuration in YAML format on the Grafana-managed alert rules form as well as Grafana-managed provisioned and non-provisioned Alert Rule detail view.

Reviewer: ```suggestion
- **View YAML button**: Displays alert rule configuration in YAML format on the Grafana-managed alert rules form, as well as Grafana-managed provisioned and non-provisioned Alert Rule detail view.
```

> - **Queries and expressions**: Several improvements have been made to the display of queries and expressions, including making Add expression a dropdown and moving Conditions to the header.

Reviewer: ```suggestion
- **Queries and expressions**: Several improvements have been made to the display of queries and expressions, including making **Add expression** a drop-down and moving **Conditions** to the header.
```

> - **Min interval option**: Improves control over query costs and performance by enabling you to adjust the minimum resolution of the data used in your alerting queries.

Reviewer: ```suggestion
- **Min interval option**: Improves control over query costs and performance by allowing you to adjust the minimum resolution of the data used in your alerting queries.
```

> - **Min interval option**: Improves control over query costs and performance by allowing you to adjust the minimum resolution of the data used in your alerting queries.

Reviewer: Not sure if alerting should be capitalized here.
```suggestion
- **Min interval option**: Improves control over query costs and performance by allowing you to adjust the minimum resolution of the data used in your alerting queries.
```

Reviewer 10: nope it can stay lower as it's not referring to the product specifically but generically to alerting functionality

- **In-app guidance for alert rule creation**: Learn about how to create your alert rules interactively with in-app guidance for additional context and links out to our Alerting documentation.
- **Support for toggling common labels**: Toggle between showing or hiding labels for each individual alert instance.

### Contact points

We’ve made the following changes to contact points.

#### Additional contact points for external Alertmanager

> Adds support for the following contact points when using an external Alertmanager:

Reviewer: ```suggestion
We've added support for the following contact points when using an external Alertmanager:
```

Reviewer 10: again, fine for me. interesting though -- i'm used to conventions that always use verbs to avoid extra words like you can, we have etc. because this is stating the obvious...is our convention to use you can, we have, etc.?

Reviewer: If this were a stand-alone section, I'd tend to agree, but overall the style of this document tends to lean a teeny bit more marketing in style, so it's less concise.

- WeChat
- Amazon SNS
- Telegram
- Cisco Webex Teams

#### Contact point provisioning file export

> Facilitates file provisioning and maintenance for contact points.This feature implements the provisioning API export endpoints for exporting contact points as well as adding export buttons to the contact point list in the UI.

Reviewer: ```suggestion
Facilitates file provisioning and maintenance for contact points. This feature implements the provisioning API export endpoints for exporting contact points as well as adding export buttons to the contact point list in the UI.
```

> Facilitates file provisioning and maintenance for contact points. This feature implements the provisioning API export endpoints for exporting contact points as well as adding export buttons to the contact point list in the UI.

Reviewer: ```suggestion
This update facilitates file provisioning and maintenance for contact points. The feature implements the provisioning API export endpoints for exporting contact points as well as adding export buttons to the contact point list in the UI.
```

### Notification policies

We’ve made the following changes to notification policies.

#### Notification policy provisioning file export

> Facilitates file provisioning and maintenance for notification policies.This feature implements the provisioning API export endpoints for exporting notification policies as well as adding an export button to the root notification policy in the UI.

Reviewer: ```suggestion
Facilitates file provisioning and maintenance for notification policies. This feature implements the provisioning API export endpoints for exporting notification policies as well as adding an export button to the root notification policy in the UI.
```

> Facilitates file provisioning and maintenance for notification policies. This feature implements the provisioning API export endpoints for exporting notification policies as well as adding an export button to the root notification policy in the UI.

Reviewer: ```suggestion
This update facilitates file provisioning and maintenance for notification policies. The feature implements the provisioning API export endpoints for exporting notification policies as well as adding an export button to the root notification policy in the UI.
```

Reviewer 10: fine for me. again though-this adds extra words that we don't necessarily need as verb is always more direct...but i'm happy to be consistent and go with the style we decide on:)

### Alert management

We’ve made the following changes to alert management.

#### Support for timezones in mute timings

> Adds support for different time zones and locations as well as a visual selector for week days, improvement to loading and error handling, and better validation for time ranges.

Reviewer: ```suggestion
We've added support for different time zones and locations as well as a visual selector for week days, made improvements to loading and error handling, and provided better validation for time ranges.
```

Reviewer 10: fine

To learn more, refer to our [Geomap network layer documentation]({{< relref "../panels-visualizations/visualizations/geomap/#network-layer-beta" >}}).

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-geomap-network-layer-v2.png" max-width="750px" caption="Geomap Network layer" >}}

### Heatmap visualizations now support data links

_Generally available in all editions of Grafana_

<!-- Nathan Marrs -->

You can now add data links to Heatmap visualizations. This allows you to add links to other dashboards, panels, or external URLs that are relevant to the data in your heatmap. We're pleased to highlight that this feature was a community contribution.

To learn more, refer to both our [Heatmap documentation]({{< relref "../panels-visualizations/visualizations/heatmap/" >}}) and our [Configure data links documentation]({{< relref "../panels-visualizations/configure-data-links/" >}}).

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-heatmap-datalinks.png" max-width="750px" caption="Heatmap datalink support" >}}

> 

Reviewer: ```suggestion

## Activate draft reports

<!-- Robert Horvath -->

_Generally available in Grafana Enterprise, Cloud Free, Cloud Pro, and Cloud Advanced_

You can now use the resume and pause report functionality to activate draft reports that have all the required fields filled in.

To learn more, refer to our [Create and manage reports documentation]({{< relref "../dashboards/create-reports" >}}).

```

_Available in public preview in all editions of Grafana_

Previously, InfluxDB backend mode was available, however, there were compatibility issues that needed to be addressed. In this release, we've addressed these issues and
promoted this feature from experimental to public preview. In the future, backend mode will be the default, and we'll deprecate frontend mode. To try backend mode, enable the `influxdbBackendMigration` feature toggle. If you’re using Grafana Cloud and would like to enable this feature, please contact customer support.

## Explore

### Logs: Choose which fields to display in a log line

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

When you're browsing logs in Explore, you can now click the eye icon within a row to replace the log line's contents with the value of one or more of the log fields or labels. This is helpful for scanning through your logs.

{{< figure src="/media/docs/grafana/log-field-picker-10-1.gif" max-width="750px" caption="Log rows menu" >}}

### Logs: Improved rendering performance of log lines

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

> Browsing log lines is faster than ever, after a series of performance optimizations to log-related components.

Reviewer: ```suggestion
After a series of performance optimizations to log-related components, browsing log lines is faster than ever.
```

Learn more in our [Angular support deprecation documentation]({{< relref "../developers/angular_deprecation/" >}}).

### Deprecated provisioning of data sources with invalid UIDs

<!-- Giuseppe Guerra, Plugins Platform -->

_Generally available in all editions of Grafana_

Grafana now logs an error when provisioning data sources with invalid UIDs. A valid UID is a combination of a-z, A-Z, 0-9 (alphanumeric), `-` (dash), and `_` (underscore), with a maximum length of 40 characters.

Provisioning data sources with invalid UIDs will be removed in future versions of Grafana, and will return an error instead.

## Subfolders: folder selection

<!-- Zsofia Komaromi -->

_Available in public preview in all editions of Grafana_

When saving or moving a dashboard, you can now see the full folder tree when selecting the destination folder.

To get started creating subfolders, enable the `nestedFolders` feature toggle. We recommend that you enable this feature only on test or development instances, rather than in production environments.

{{< figure src="/media/docs/grafana/screenshot-grafana-10.1-subfolders-folder-picker.png" max-width="750px" caption="Selecting a folder in Grafana" >}}

> ## Activate draft reports
> 

Reviewer: ```suggestion
```

> <!-- Robert Horvath -->
> 
> _Generally available in Grafana Enterprise, Cloud Free, Cloud Pro, and Cloud Advanced_
> 
> You can now use the resume and pause report functionality to activate draft reports that have all the required fields filled in.
> 
> To learn more, refer to our [Create and manage reports documentation]({{< relref "../dashboards/create-reports" >}}).

Reviewer: ```suggestion
```

To learn more, refer to our [Geomap network layer documentation]({{< relref "../panels-visualizations/visualizations/geomap/#network-layer-beta" >}}).

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-geomap-network-layer-v2.png" max-width="750px" caption="Geomap Network layer" >}}

### Heatmap visualizations now support data links

_Generally available in all editions of Grafana_

<!-- Nathan Marrs -->

You can now add data links to Heatmap visualizations. This allows you to add links to other dashboards, panels, or external URLs that are relevant to the data in your heatmap. We're pleased to highlight that this feature was a community contribution.

To learn more, refer to both our [Heatmap documentation]({{< relref "../panels-visualizations/visualizations/heatmap/" >}}) and our [Configure data links documentation]({{< relref "../panels-visualizations/configure-data-links/" >}}).

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-heatmap-datalinks.png" max-width="750px" caption="Heatmap datalink support" >}}

> ## Activate draft reports

Reviewer: ```suggestion
### Activate draft reports
```

### Transformations redesign

<!-- Catherine Gui -->

_Available in public preview in all editions of Grafana_

The transformations tab has an improved user experience and visual redesign. Now, transformations are categorized, and each transformation type has an illustration to help you choose the right one.

{{< figure src="/media/docs/grafana/screenshot-grafana-10-1-transformations.png" max-width="750px" caption="Transformations redesign" >}}

### Format Time transformation

<!-- Kyle Cunningham -->

_Available in public preview in all editions of Grafana_

When working with date and time data, it can be useful to have different time formats. With the new Format Time transformation, you can convert any time format to any other one supported by [Moment.js](https://momentjs.com/docs/#/displaying/). When used in conjunction with the _Group by_ transformation, this can also be used to bucket days, weeks, and other time windows together.

{{< figure src="/media/docs/grafana/format-time-10-1.gif" max-width="750px" caption="Format time transformation" >}}

### Join by fields transformation outer join (tabular) option

<!-- Brendan O'Handley -->

> 

Reviewer: ```suggestion

_Generally available in all editions of Grafana_

```

Currently, you can add one or more of the following filters:

- Service Name
- Span Name
- Duration
- Tags (which includes tags, process tags, and log fields)

To only show the spans you've matched, you can enable the **Show matches only** toggle.

Learn more about span filtering in our [Tempo data source documentation]({{< relref "../datasources/tempo/#span-filters" >}}).

{{< figure src="/media/docs/tempo/screenshot-grafana-tempo-span-filters-v10-1.png" max-width="750px" caption="Traces span filtering" >}}

### Configuration page redesign for Loki and Elasticsearch

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_

The Loki and Elasticsearch data source configuration pages have been redesigned to make getting started and setting up data sources as simple and easy to understand as possible. You can now find new subsections with links to documentation pages, as well as tooltips to assist you with configuring and customizing data sources.

> ### Loki query splitting
> 
> <!-- Matías Wenceslao Chomicki -->
> 
> _Generally available in all editions of Grafana_
> 
> In response to different query performance scenarios, we've implemented query splitting. Now, queries that request more than a day of data are split into sub-requests, each with a duration of one day. For example, requesting seven days of logs will produce seven requests of one day.
> 

Reviewer 6: We need to remove this from What's new because we have to rollback this release.

```suggestion
### Loki query splitting

<!-- Matías Wenceslao Chomicki -->

_Generally available in all editions of Grafana_
```

Reviewer: ```suggestion
```

To avoid overriding manually set roles, enable the `skip_org_role_sync` option in the Grafana configuration for your OAuth provider before the user logs in for the first time.

### Preventing manual role updates for externally synced roles

<!-- Ieva Vasiljeva, AuthNZ -->

_Generally available in Grafana Open Source, Enterprise, and Cloud._

This change impacts all instances that use an external authentication provider and have [role mapping]({{< relref "../setup-grafana/configure-security/planning-iam-strategy/#role-sync" >}}) enabled.

Currently, it is possible to manually update a user's organization role (Viewer, Editor or Admin) even if this role is managed by an external authentication provider.

> This means that roles can be manually set for the duration of a user's session, but are overridden by the external authentication provider the next time when the user logs in.
> If `onlyExternalOrgRoleSync` feature toggle is enabled, manual role updates for externally managed roles are not allowed.

Reviewer: ```suggestion
This means that roles can be manually set for the duration of a user's session, but are overridden by the external authentication provider the next time the user logs in.
If the `onlyExternalOrgRoleSync` feature toggle is enabled, manual role updates for externally managed roles are not allowed.
```
